### PR TITLE
feat(app): Enable download robot logs in advanced settings

### LINKED ISSUE #1727: Debug and Admin: Download Logs
As a robot user, I would like to be able to download robot logs from my run app, so that I can share them with the Opentrons support team.

## Acceptance Criteria
- Button on Robot Settings page in Run App to download logs
- Clicking button downloads robot logs to user's laptop/desktop

## Design
- TBD (keeping it simple for now)
- @pantslakz

### PR DESCRIPTION
**Pull request by Kadee80**
## overview

This PR closes #1727. Download Logs button is now enabled at the bottom of `RobotSettingsPage` in the `AdvancedSettingsCard`

## changelog

- feat(app):Enable download robot logs in advanced settings 

## review requests

Standard code review.

NOTES:
- Virtual Smoothie does not have logs but will prompt anyway
- Some robots in the office do not have `api.logs` if they have not run a protocol in a while
- Currently, if the logs do not exisit the user is prompted to `save as` anyway and but no file will download
- @Laura-Danielle is going to open an issue/PR to create blank log files to avoid this confusing situation.

### CODE REVIEW

File: app-shell/lib/robot-logs.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// download robot logs manager
+
+module.exports = {
+  downloadRobotLogs (dispatch, mainWindow) {
```
Comment (mcous) [inline app-shell/lib/robot-logs.js]: Can we match the existing naming convention on this and call the function `registerRobotLogs`?

File: app-shell/lib/robot-logs.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// download robot logs manager
```
Comment (mcous) [inline app-shell/lib/robot-logs.js]: Post #2013, this file should be `app-shell/src/robot-logs.js` and it should use `export function`, not `module.exports`

File: app/src/http-api-client/health.js [hunk 3]
```diff
@@ -16,6 +16,7 @@ type HealthResponse = {
   name: string,
   api_version: string,
   fw_version: string,
+  logs: Array<string>,
```
Comment (mcous) [inline app/src/http-api-client/health.js]: Let's make this a `?Array<string>` so we cover our bases with robot API servers that may be out of date

File: app/src/components/RobotSettings/AdvancedSettingsCard.js [hunk 4]
```diff
@@ -64,7 +66,7 @@ function AdvancedSettingsCard (props: Props) {
       <LabeledButton
         label='Download Logs'
         buttonProps={{
-          disabled: true,
```
Comment (mcous) [inline app/src/components/RobotSettings/AdvancedSettingsCard.js]: This PR is small enough that I think we should sneak this in: let's add get robot health to `mapStateToProps` here just so we can set `disabled: true` if `health.response.logs` does not exist. We should keep the thunk action exactly the same, though

File: app-shell/lib/robot-logs.js (hunk 2 above)
Comment (Kadee80) [reply app-shell/lib/robot-logs.js]: ok cool. rebase didnt catch it. will move and update

File: app-shell/lib/robot-logs.js (hunk 1 above)
Comment (Kadee80) [reply app-shell/lib/robot-logs.js]: 👍

File: app-shell/lib/robot-logs.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+'use strict';
```
Comment (mcous) [inline app-shell/lib/robot-logs.js]: This file still needs to be removed from git tracking

**Pull request merged**